crates/matching-engine/src/cfmm/uniswap/mod.rs

```diff
@@ -1,8 +1,32 @@
+use alloy_primitives::{aliases::I24, I256};
+
 pub mod pool;
+pub mod pool_data_loader;
 pub mod pool_manager;
 pub mod pool_providers;
 pub mod tob;
 
+fn i32_to_i24(val: i32) -> I24 {
+    let mut bytes = [0u8; 3];
+    let value_bytes = val.to_be_bytes();
+    bytes[..].copy_from_slice(&value_bytes[1..]);
+    I24::from_be_bytes(bytes)
+}
+
+fn i128_to_i256(value: i128) -> I256 {
+    let mut bytes = [0u8; I256::BYTES];
+    let value_bytes = value.to_be_bytes();
+    bytes[16..].copy_from_slice(&value_bytes);
+    I256::from_be_bytes(bytes)
+}
+
+fn i256_to_i128(value: I256) -> i128 {
+    let mut bytes = [0u8; 16];
+    let value_bytes: [u8; I256::BYTES] = value.to_be_bytes();
+    bytes.copy_from_slice(&value_bytes[16..]);
+    i128::from_be_bytes(bytes)
+}
```

Comment:
few things, mostly related to comments above but check twos compliment here and also if we could have a panic on size overflow that would be great

Reply:
It seems that unsigned do. Signed don't? 

```
error[E0599]: no method named `to` found for struct `alloy_primitives::Signed` in the current scope
   --> crates/matching-engine/src/cfmm/uniswap/pool_data_loader.rs:401:58
    |
401 |             liquidity_delta: modify_event.liquidityDelta.to::<i128>()
    |                                                          ^^
    |
help: there is a method `into` with a similar name
    |
401 |             liquidity_delta: modify_event.liquidityDelta.into::<i128>()
    |                                                          ~~~~
```

Reply:
rip